app/decorators/controllers/solidus_gdpr/spree/api/users_controller_decorator.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module SolidusGdpr
+  module Spree
+    module Api
+      module UsersControllerDecorator
+        def emails
+          users_scope = model_class.accessible_by(current_ability, :read)
+          orders_scope = ::Spree::Order.all
+
+          users_scope = users_scope.ransack(params[:q]).result
+          orders_scope = orders_scope.ransack(params[:q]).result(distinct: true)
+
+          scope = (users_scope | orders_scope).map(&:email).uniq.reject do |email|
+            email.split('@').last == SolidusGdpr.configuration.erased_email.split('@').last
```

Comment:
I want to prevent that the email select returns also the erased emails. 
In my option, there are two solutions:

- Add the `erased` column to the user and order models to filter them.
- Don't use the proc or the method, in this case, I can use the same method used now.

Reply:
The erased emails wouldn't be part of the search results anyway. We should start searching for emails after a minimum character threshold is reached (let's say a min of 3 chars), so that we don't perform useless searches in the DB. This will prevent erased emails from showing up, since they'll just be made of random characters.